src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -0,0 +1,180 @@
+        # pylint: enable=consider-using-enumerate
+
+        self.log.info("Job-ID: %s", self.job_id)
+        self.log.info("Test PID: %s", os.getpid())
+
+        self.basepath = None
+        self.orterun = None
+        self.tmp = None
+        self.server_group = None
+        self.daosctl = None
+        self.context = None
+        self.pool = None
+        self.container = None
+        self.hostlist = None
+        self.hostlist_servers = None
+        self.hostfile_servers = None
+        self.d_log = None
+        self.uri_file = None
+        self.fault_file = None
+
+    # pylint: disable=invalid-name
+    def cancelForTicket(self, ticket):
+        ''' Skip a test due to a ticket needing to be completed '''
+        return self.cancel("Skipping until {} is fixed.".format(ticket))
+    # pylint: enable=invalid-name
+
+class TestWithoutServers(Test):
+    '''
+    Run tests without DAOS servers
+
+    :avocado: recursive
+    '''
+    def setUp(self):
+        super(TestWithoutServers, self).setUp()
+        # get paths from the build_vars generated by build
+        with open('../../../.build_vars.json') as build_vars:
+            build_paths = json.load(build_vars)
+        self.basepath = os.path.normpath(build_paths['PREFIX']  + "/../")
+        self.prefix = build_paths['PREFIX']
+        self.tmp = self.prefix + '/tmp'
+        self.daos_test = self.basepath + '/install/bin/daos_test'
+        self.orterun = self.basepath + '/install/bin/orterun'
+        self.daosctl = self.basepath + '/install/bin/daosctl'
+
+        # setup fault injection, this MUST be before API setup
+        fault_list = self.params.get("fault_list", '/run/faults/*/')
+        if fault_list:
+            # not using workdir because the huge path was messing up
+            # orterun or something, could re-evaluate this later
+            tmp = os.path.join(self.basepath, 'install', 'tmp')
+            self.fault_file = fault_config_utils.write_fault_file(tmp,
+                                                                  fault_list,
+                                                                  None)
+            os.environ["D_FI_CONFIG"] = self.fault_file
+
+        self.context = DaosContext(self.prefix + '/lib/')
+        self.d_log = DaosLog(self.context)
+
+    def tearDown(self):
+        super(TestWithoutServers, self).tearDown()
+
+        if self.fault_file:
+            os.remove(self.fault_file)
+
+class TestWithServers(TestWithoutServers):
+    '''
+    Run tests with DAOS servers
+
+    :avocado: recursive
+    '''
+    def __init__(self, *args, **kwargs):
+        super(TestWithServers, self).__init__(*args, **kwargs)
+
+        self.agent_sessions = None
+
+    def setUp(self):
+        super(TestWithServers, self).setUp()
+
+        self.server_group = self.params.get("server_group", '/server/',
+                                            'daos_server')
+        self.hostlist = self.params.get("test_machines", '/run/hosts/*')
```

Comment:
Since we are making this a standard pattern perhaps update "test_machines" to daos_servers or something since as we add more complex tests there will be more categories of test machines and in this case it is specifically servers that we are configuring.

Reply:
I think this is a bigger can of worms that simply trying to do a `s/test_machines/daos_servers/g`.  I think there is some lack of consistency with how machines are specified across too many tests to try to make this change now wholesale.